engine/transaction/outlines/internal/inputs/selector_test.go

```diff
@@ -0,0 +1,158 @@
+			selectBy: selectBy{
+				satoshis:            10,
+				txSizeWithoutInputs: 116,
+			},
+			expectToSelectInputs: []int{0, 1},
+		},
+		"select inputs that covers outputs and fee for more data": {
+			selectBy: selectBy{
+				satoshis:            9,
+				txSizeWithoutInputs: uint64(fixtures.DefaultFeeUnit.Bytes + 1 - fixtures.EstimatedUnlockingScriptSizeForP2PKH),
+			},
+			expectToSelectInputs: []int{0, 1},
+		},
+		"select inputs when size is equal to fee unit bytes": {
+			selectBy: selectBy{
+				satoshis:            9,
+				txSizeWithoutInputs: uint64(fixtures.DefaultFeeUnit.Bytes - fixtures.EstimatedUnlockingScriptSizeForP2PKH),
+			},
+			expectToSelectInputs: []int{0},
+		},
+	}
+	for name, test := range singleSelectTests {
+		t.Run(name, func(t *testing.T) {
+			// given:
+			given, then, cleanup := testabilities.New(t)
+			defer cleanup()
+
+			// and: having some utxo in database
+			var ownedInputs []*database.UserUtxos
+			ownedInputs = append(ownedInputs, given.DB().HasUTXO().OwnedBySender().P2PKH().WithSatoshis(10).Stored())
+			ownedInputs = append(ownedInputs, given.DB().HasUTXO().OwnedBySender().P2PKH().WithSatoshis(10).Stored())
+			ownedInputs = append(ownedInputs, given.DB().HasUTXO().OwnedBySender().P2PKH().WithSatoshis(10).Stored())
+			ownedInputs = append(ownedInputs, given.DB().HasUTXO().OwnedBySender().P2PKH().WithSatoshis(10).Stored())
+
+			// and:
+			selector := given.NewInputSelector()
+
+			// when:
+			utxos, err := selector.SelectInputsForTransaction(context.Background(), fixtures.Sender.XPubID(), test.selectBy.satoshis, test.selectBy.txSizeWithoutInputs)
+
+			// then:
+			then.WithoutError(err).SelectedInputs(utxos).
+				ComparingTo(ownedInputs).AreEntries(test.expectToSelectInputs)
+
+		})
+	}
+
+	twiceSelectTests := map[string]struct {
+		selectBy             selectBy
+		expectToSelectInputs []int
+	}{
+		"select different inputs for second call": {
+			selectBy: selectBy{
+				satoshis:            15,
+				txSizeWithoutInputs: 116,
+			},
+			expectToSelectInputs: []int{2, 3},
+		},
+		"select already touched inputs if the amount of not touched won't fulfill required amount": {
+			selectBy: selectBy{
+				satoshis:            25,
+				txSizeWithoutInputs: 116,
+			},
+			expectToSelectInputs: []int{0, 1, 3},
+		},
+	}
+	for name, test := range twiceSelectTests {
+		t.Run(name, func(t *testing.T) {
+			// given:
+			given, then, cleanup := testabilities.New(t)
+			defer cleanup()
+
+			// and: having some utxo in database
+			var ownedInputs []*database.UserUtxos
+			ownedInputs = append(ownedInputs, given.DB().HasUTXO().OwnedBySender().P2PKH().WithSatoshis(10).Stored())
+			ownedInputs = append(ownedInputs, given.DB().HasUTXO().OwnedBySender().P2PKH().WithSatoshis(10).Stored())
+			ownedInputs = append(ownedInputs, given.DB().HasUTXO().OwnedBySender().P2PKH().WithSatoshis(10).Stored())
+			ownedInputs = append(ownedInputs, given.DB().HasUTXO().OwnedBySender().P2PKH().WithSatoshis(10).Stored())
```

Comment:
```
ownedInputs := []*database.UserUtxos{
		given.DB().HasUTXO().OwnedBySender().P2PKH().WithSatoshis(10).Stored(),
		given.DB().HasUTXO().OwnedBySender().P2PKH().WithSatoshis(10).Stored(),
		given.DB().HasUTXO().OwnedBySender().P2PKH().WithSatoshis(10).Stored(),
		given.DB().HasUTXO().OwnedBySender().P2PKH().WithSatoshis(10).Stored(),
}
```

Reply:
good point, I was thinking about other approach and forgot to make it better.